File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
s = input().strip()
sub = input().strip()

q = []

for char in s:
    q.append(char)

    if len(q) >= len(sub):
        if q[-len(sub):] == list(sub):
            del q[-len(sub):]

print("".join(q) or "FRULA")

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [str, sub] = input;

function solution(str, sub) {
  const q = [];

  for (let i = 0; i < str.length; i++) {
    q.push(str[i]);

    if (q.length >= sub.length) {
      const lastStr = q.slice(-sub.length).join("");

      if (lastStr === sub) {
        q.splice(-sub.length);
      }
    }
  }

  return q.join("") || "FRULA";
}

console.log(solution(str, sub));
